action-allowedlist/action_allowedlist/actions_parser.py

```diff
@@ -0,0 +1,136 @@
+        )
+    if not workflow_files:
+        print("Could not find workflow files in the specified directory")
+        return []
+
+    print(f"Found [{len(workflow_files)}] files in the workflows directory")
+
+    actions_in_repo = []
+
+    for workflow_file in workflow_files:
+        try:
+            workflow_content = workflow_file.read_text()
+            actions = get_actions_from_file(workflow_content, workflow_file.name)
+            actions_in_repo.extend(actions)
+        except Exception as e:
+            print(f"Error occurred while reading {workflow_file}: {e}")
+
+    return actions_in_repo
+
+
+def invoke_validate_actions(approved_path, actions_configuration):
+    print("Checking if used actions are approved")
+
+    approved = load_json_file(approved_path)
+    num_approved = 0
+    num_denied = 0
+    num_deprecated = 0
+    unapproved_outputs = []
+    approved_outputs = []
+    found = False
+
+    for action in actions_configuration:
+        print(
+            f"::debug::Processing {action['actionLink']} version {action['actionVersion']}"
+        )
+        approved_action_versions = [
+            a for a in approved if a["actionLink"] == action["actionLink"]
+        ]
+        if approved_action_versions:
+            print(
+                f"::debug::Approved Versions for {action['actionLink']}: {[a['actionVersion'] for a in approved_action_versions]}"
+            )
+        else:
+            print(
+                f"::debug::No Approved versions for {action['actionLink']} were found."
+            )
+            pass
+
+        approved_output = [
+            a
+            for a in approved_action_versions
+            if a["actionVersion"] == action["actionVersion"]
+        ]
+
+        if approved_output:
+            print(f"::debug::Output versions approved: {approved_output}")
+            approved_outputs.append(approved_output[0])
+            num_approved += 1
+
+            # Look for deprecation
+            if approved_output[0].get("deprecated", False):
+                print(f"Using a deprecated version of {action['actionLink']}")
+                num_deprecated += 1
+                found = True
+        else:
+            print(
+                f"::debug::Output versions not approved: {action['actionLink']} version {action['actionVersion']}"
+            )
+            unapproved_outputs.append(
+                f"{action['actionLink']} {action['actionVersion']}"
+            )
+            num_denied += 1
+            found = True
+
+    if num_denied > 0:
+        e = f"The following {num_denied} actions/versions were denied: {', '.join(unapproved_outputs)}"
+        print(f"::error file=actions_parser.py,title=Denied Actions::{e}")
+        return found
+    else:
+        print(f"All {num_approved} actions/versions are approved.")
+        if num_deprecated > 0:
+            print(f"Deprecated actions found: {num_deprecated}")
```

Comment:
Let's change this to a warning annotation so that it is more visible. Similar to "::error" above, but "::warning"